src/Microsoft.Extensions.Http/ITypedHttpClientFactory.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Http;
+
+namespace Microsoft.Extensions.Http
+{
+    public interface ITypedHttpClientFactory
```

Comment:
A library like Refit can implement this and **just work** with our extension methods.